packages/ui/src/components/flag/flag.native.tsx

```diff
@@ -0,0 +1,50 @@
+import { ViewStyle } from 'react-native';
+
+import { ResponsiveValue } from '@shopify/restyle';
+
+import { Theme } from '../../theme-native';
+import { Box } from '../box/box.native';
+import type { FlagAlignment } from './flag.shared';
+
+export interface FlagProps {
+  align?: FlagAlignment;
+  children: React.ReactNode;
+  img?: React.ReactNode;
+  reverse?: boolean;
+  spacing?: ResponsiveValue<keyof Theme['spacing'], Theme['breakpoints']>;
+  boxProps?: ViewStyle;
+}
+
+const getFlagAlignment = (align: FlagAlignment) => {
+  switch (align) {
+    case 'top':
+      return 'flex-start';
+    case 'middle':
+      return 'center';
+    case 'bottom':
+      return 'flex-end';
+  }
+};
+
+export function Flag({
+  align = 'middle',
+  img,
+  children,
+  reverse = false,
+  spacing = '3',
+  boxProps = {},
```

Comment:
@edgarkhanzadian : I have a Q about how I want this component to be extensible. 

On web we have `Flag` which extends `FlexProps`. In this PR I try to achieve the same thing by allowing `style={boxStyles}` so that we could pass extra props in if we need to: 

```
<Flag marginLeft={X
```

Is this the right pattern to follow?